### PR TITLE
Bugfix values function

### PR DESCRIPTION
**Pull request by mahasadhu**
Values function in fluent API not calling the inner / callback / anonymous function sent to it as parameter

### CODE REVIEW

File: src/app/Library/CrudPanel/CrudFilter.php [hunk 1]
```diff
@@ -285,7 +285,7 @@ public function label($value)
      */
     public function values($value)
     {
-        $this->values = $value;
+        $this->values = is_callable($value) ? $value() : $value;
```
Comment (pxpm) [inline src/app/Library/CrudPanel/CrudFilter.php:288]: We had trouble using this kind of check in the past, who knows what values could endup there :smiley: 

 Just to make sure we don't get into the same hole I would go here with: 

`$this->values = (!is_string($value) && is_callable($value)) ? $value() : $value;`

If the string is a php core function we know what is going to happen :+1:

**Review by pxpm — CHANGES_REQUESTED** [review]
Apart from my comment I very much welcome this PR. Thank you @mahasadhu 

This must have slipped while we refactored Filters into classes. So sorry for the trouble. 

If you don't have time to make the suggested changes let us know so we can finish this up and merge the PR. 

Wish you the best,
Pedro

**Comment by mahasadhu** [comment]
@pxpm Hi Pedro, thanks for checking 😃 

Sorry I did not know that. I've pushed some updates to the code.

Another idea, we can just close this PR and update the docs from this:
https://backpackforlaravel.com/docs/4.1/crud-fluent-syntax
image

to, for example, this:
```
            CRUD::filter('department')
                ->type('select2')
                ->label(trans('backpack::base.job_category'))
                ->values(\App\Models\Department::all()->pluck('name', 'id')->toArray())
                ->whenActive(function($value) { // if the filter is active
                    $this->crud->query->whereHas('project.department', function ($query) use ($value) {
                        $query->where('id', $value);
                    });
                });
```

Because (CMIIW) it seems no one has realized this errors yet. But it's up to you though, just a thought.

Thank you!

**Comment by pxpm** [comment]
Hello @mahasadhu thank you very much once again :+1:

The refactor to use Fluent api for Filters happened not so long time, so that's why I think not so many people switch from old array definition to the new fluent syntax. 

The array definition supports anonymous functions in value key, this was just a mistake we let slip when refactoring to the Fluent API. 

With your PR merged the documentation should be ok ? 

Best,
Pedro

**Comment by mahasadhu** [comment]
No problem at all @pxpm

Ah I see...
Yes it should be ok, no need to update the docs :smiley

Thank you

**Comment by tabacitu** [comment]
Thank a lot @mahasadhu & @pxpm . You're right, of course. 
Merged, will be tagged later today, and be available as a patch update with `composer update`.

Keep 'em coming! Cheers.

**Pull request merged**